app/src/main/java/me/ccrama/redditslide/Fragments/WikiPage.java

```diff
@@ -38,66 +61,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         ref.setProgressViewOffset(false,
                 Constants.SINGLE_HEADER_VIEW_OFFSET - Constants.PTR_OFFSET_TOP,
                 Constants.SINGLE_HEADER_VIEW_OFFSET + Constants.PTR_OFFSET_BOTTOM);
-
         ref.post(new Runnable() {
             @Override
             public void run() {
                 ref.setRefreshing(true);
             }
         });
-        new AsyncTask<Void, Void, Void>() {
-            String text;
+    }
 
+    private void setUpWebView() {
+        webView.setWebViewClient(new WebViewClient() {
             @Override
-            protected Void doInBackground(Void... params) {
-                try {
-                    text = ((Wiki) getActivity()).wiki.get(subreddit, title).getDataNode().get("content_html").asText();
-                } catch(Exception ignored){
-
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                if (url.toLowerCase().startsWith(wikiUrl.toLowerCase()) && listener != null) {
+                    String pagePiece = url.toLowerCase().replace(wikiUrl.toLowerCase(), "")
+                            .split("/")[0]
+                            .split("#")[0];
+                    listener.embeddedWikiLinkClicked(pagePiece);
+                } else {
+                    OpenRedditLink.openUrl(getContext(), url, true);
                 }
-                return null;
+                return true;
             }
 
```

Comment:
Think I found a way to get rid of the white flashes, adding:

```java
            @Override
            public void onPageStarted(WebView view, String url, Bitmap favicon) {
                webView.setBackgroundColor(Color.TRANSPARENT);
                super.onPageStarted(view, url, favicon);
            }
```

took care of it for me. Might be more performant to set the correct background colour rather than transparent but I couldn't figure out which class to use for that. Seems to be some duplicated things between Palette.java and ColorPreferences.java